### PR TITLE
Merging deconvolution examples (again) #2512

### PR DESCRIPTION
**Pull request by alexandrejaguar**
## Description
Merge two deconvolution (Wiener and Richardson-Lucy) examples.
Sorry about the last commit. I made something _really_ wrong again...

## References
Related to issue #2506.
[1] http://scikit-image.org/docs/dev/auto_examples/filters/plot_restoration.html#sphx-glr-auto-examples-filters-plot-restoration-py
[2] http://scikit-image.org/docs/dev/auto_examples/filters/plot_deconvolution.html#sphx-glr-auto-examples-filters-plot-deconvolution-py

### CODE REVIEW

File: doc/examples/filters/plot_deconvolution.py [hunk 1]
```diff
@@ -2,54 +2,106 @@
...
-optical system. The blurred image is sharpened through a number of
-iterations, which needs to be hand-tuned.
-
-.. [1] William Hadley Richardson, "Bayesian-Based Iterative
-       Method of Image Restoration",
-       J. Opt. Soc. Am. A 27, 1593-1607 (1972), DOI:10.1364/JOSA.62.000055
-
-.. [2] http://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution
+In this example, Wiener, unsupervised Wiener, and Richardson-Lucy
+algorithms are used to deconvolve an image.
```
Comment (sciunto) [inline doc/examples/filters/plot_deconvolution.py]: Blank like before the title?

Comment (alexandrejaguar) [reply doc/examples/filters/plot_deconvolution.py]: Yep. What about the new paragraphs?

File: doc/examples/filters/plot_deconvolution.py [hunk 2]
```diff
@@ -2,54 +2,106 @@
...
+First, a noisy version of an image is deconvolved using Wiener
+and unsupervised Wiener algorithms. These algorithms are based on
+linear models that cannot restore sharp edges as much as non-linear
+methods (like TV restoration), but are faster.
+The Wiener deconvolution algorithm is an inverse filter based on an
+impulse response function (also called point-spread function, or PSF), a
+filter to penalize the restored image frequencies (regularization), and
+a parameter which tunes the balance between the data and the
+regularization.
+The unsupervised Wiener algorithm has self-tuned regularization
```
Comment (soupault) [inline doc/examples/filters/plot_deconvolution.py]: Same here: the functional is not explicit, and the regularization impact is unclear.
Does mentioning Gibbs sampling clarify things here?

**Comment by soupault** [comment]
Whoa! Sorry, didn't mean to close this PR! I was only cleaning the upstream in accordance with the following issue - https://github.com/scikit-image/scikit-image/issues/3448 . Apparently, your branch was created in upstream...

**Comment by alexandrejaguar** [comment]
@soupault I don't know what I'm doing! :sweat_smile: Formatted the PC these days, things here are a mess. I'll get to that in a few, ok?

**Pull request closed without merging**